labs/a5.md

```diff
@@ -0,0 +1,336 @@
+
+## Sysadmin commands
+
+### ifupdown
+`ifupdowm` is a simple suite of commands for interacting with network interfaces.  The two commands you'll be using most are `ifup` and `ifdown` which are relatively self-explanatory.  `ifup` brings and interface up and vice versa for `ifdown`.  These two commands should be your de facto commands for bringing interfaces up or down since using these commands loads configurations defined in **/etc/network/interfaces**.
+
+### mtr
+`mtr` is a command that combines the functionality of `traceroute` with that of `ping`. Take a look a look at [this article][mtr-primer] for a good primer for using `mtr` and interpreting its output.
+
+[mtr-primer]: https://linode.com/docs/networking/diagnostics/diagnosing-network-issues-with-mtr/
+
+### iptables
+In favor of not reinventing the wheel please check out these excellent and pretty short articles by **DigitalOcean**, who sponsored this semester's offering of the decal by supplying us with VMs.
+1. [An Introduction to iptables][intro]
+2. [Adding rules][firewall]
+3. [Deleting rules][delet]
+4. [Common rules and tips][tips]
+
+[intro]: https://www.digitalocean.com/community/tutorials/how-the-iptables-firewall-works
+[firewall]: https://www.digitalocean.com/community/tutorials/how-to-set-up-a-firewall-using-iptables-on-ubuntu-14-04
+[delet]: https://www.digitalocean.com/community/tutorials/how-to-list-and-delete-iptables-firewall-rules
+[tips]: https://www.digitalocean.com/community/tutorials/iptables-essentials-common-firewall-rules-and-commands
+
+# Exercises
+**Checkoff form can be found [here][checkoff].  Please read the sections below as well, there are instructions for setting up your environment.**
+
+[checkoff]: https://goo.gl/forms/OCxyAu1otMn91Oa13
+
+## Riddle Me This
+1. Is the result of running `ping` enough to determine whether or not you can reach a server, i.e. if you can ping a server you can reach a server and vice versa.  Justify.
+2. Describe the `tcp_syncookies` `sysctl` option.  How can we toggle this value on, and when would we want this on?
+3. Where does `dhclient` store old leases?
+4. Please write a few stanzas that configures an interface called `test` and configures it so that it is brought up on boot and given the following address: `192.168.13.37/16`.  Where should this snippet be placed?
+5. Assume the following information:
+* **/etc/resolv.conf** file has 3 `nameserver` entries and a `options timeout:1` entry.
+* A successful DNS response takes 20 ms.
+
+  You need to add the `retry:n` option so that you retry a query as many times as possible but the total time to resolve a name, irrelevant of success or failure, takes less than **5** seconds.  What should the value of n be?
+6. If we preferred name resolution be done dynamically rather than using static entries in **/etc/hosts** what file do we need to edit and what is the line we should add?
```

Comment:
This gets renumbered to `1` since it unindents the previous block when rendering the markdown. Try indenting the line above (starting with `You need to add the`) to 4 spaces and that should fix it.

Reply:
shifted questions around hopefully that fixes it